python/examples/explore.py

```diff
@@ -0,0 +1,295 @@
+"""
+Chemiscope.explore example
```

Comment:
Do we want a single example with all different featurization schemes, or do we want one example with the default and then other examples with other schemes?

@bananenpampe @ceriottm

Reply:
I think the idea here is to demonstrate the usage of chemiscope.explore more than to show cool visualizations - so I'd have enough examples to show how to use it in the basic mode, and how to create custom featurizers.

Reply:
In this case we don't need to use `fetch_dataset` and instead always use a small set of frames. We also don't have to make an example with MACE/t-SNE, since custom featurizers can be demonstrated with the existing dscribe + KPCA example.

I would only keep two examples:

- basic, do nothing special
- custom featurizer + custom properties

----

There was a discussion of having more featurizers such as MACE/t-SNE in example as an easy to copy/paste version of the code for people who aren't familiar with this kind of things, should we move these to a separate (set of) examples?

Reply:
Where else would be a good place to show these examples?
Should there be a chemiscope-gallery  that shows/explains them?

Reply:
These could still be in the same place IMO, but split into "explaining how to use chemiscope.explore in general" and "specific applications of chemiscope.explore" (one example file per application)

Reply:
Okay, I have separated it to generic and specific applications, let me know if I need change anything else